elasticsearch_dsl/faceted_search.py

```diff
@@ -130,6 +130,7 @@ class DateHistogramFacet(Facet):
     agg_type = 'date_histogram'
 
     DATE_INTERVALS = {
+        'year': lambda d: d.replace(year=d.year+1),
```

Comment:
This logic actually breaks for Feb 29, could you perhaps change it to a similar logic as `month`? Thanks!